process/slash/detector/multipleHeaderProposalsDetector_test.go

```diff
@@ -0,0 +1,452 @@
+package detector_test
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	mockEpochStart "github.com/ElrondNetwork/elrond-go/epochStart/mock"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/block/interceptedBlocks"
+	"github.com/ElrondNetwork/elrond-go/process/mock"
+	"github.com/ElrondNetwork/elrond-go/process/slash"
+	"github.com/ElrondNetwork/elrond-go/process/slash/detector"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewMultipleHeaderProposalsDetector(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		args        func() (sharding.NodesCoordinator, process.RoundHandler, uint64)
+		expectedErr error
+	}{
+		{
+			args: func() (sharding.NodesCoordinator, process.RoundHandler, uint64) {
+				return nil, &mock.RoundHandlerMock{}, detector.CacheSize
+			},
+			expectedErr: process.ErrNilShardCoordinator,
+		},
+		{
+			args: func() (sharding.NodesCoordinator, process.RoundHandler, uint64) {
+				return &mock.NodesCoordinatorMock{}, nil, detector.CacheSize
+			},
+			expectedErr: process.ErrNilRoundHandler,
+		},
+		{
+			args: func() (sharding.NodesCoordinator, process.RoundHandler, uint64) {
+				return &mock.NodesCoordinatorMock{}, &mock.RoundHandlerMock{}, detector.CacheSize
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, currTest := range tests {
+		_, err := detector.NewMultipleHeaderProposalsDetector(currTest.args())
+		require.Equal(t, currTest.expectedErr, err)
+	}
+}
+
+func TestMultipleHeaderProposalsDetector_VerifyData_CannotCastData_ExpectError(t *testing.T) {
+	t.Parallel()
+
+	sd, _ := detector.NewMultipleHeaderProposalsDetector(&mock.NodesCoordinatorMock{}, &mock.RoundHandlerMock{}, detector.CacheSize)
+	res, err := sd.VerifyData(&testscommon.InterceptedDataStub{})
+
+	require.Nil(t, res)
+	require.Equal(t, process.ErrCannotCastInterceptedDataToHeader, err)
+}
+
+func TestMultipleHeaderProposalsDetector_VerifyData_CannotGetProposer_ExpectError(t *testing.T) {
+	t.Parallel()
+
+	expectedErr := errors.New("cannot get proposer")
+	sd, _ := detector.NewMultipleHeaderProposalsDetector(&mockEpochStart.NodesCoordinatorStub{
```

Comment:
can you move the instantiation of nodesCoordinator outside of the parameter list ? (it's harder to follow when reviewing)